src/jvmMain/kotlin/net/kyori/adventure/webui/jvm/minimessage/MiniMessage.kt

```diff
@@ -167,6 +171,27 @@ public fun Application.miniMessage() {
                 call.respondText(root.toString())
             }
 
+            post(URL_MINI_SHORTEN) {
+                val structure = Serializers.json.tryDecodeFromString<Combined>(call.receiveText())
+                val code = BytebinStorage.bytebinStore(structure ?: return@post)
+                if (code != null) {
+                    call.respondText(code)
+                } else {
+                    call.response.status(HttpStatusCode.InternalServerError)
+                }
+            }
+
+            get(URL_MINI_SHORTEN) {
+                val code = call.parameters["code"]
+                val structure = BytebinStorage.bytebinLoad(code ?: return@get)
+                if (structure != null) {
+                    // Pretty sure this is pointlessly decoding from json and then re-encoding to the same thing
```

Comment:
Well, this does have the added benefit of ensuring it's not an invalid bytebin paste right?

Reply:
Yes, but technically it doesn't need to be converted *back* to json

Reply:
Ah right yeah that does make sense. Well this is fine for now ig